File: go/rna-transcription/rna_transcription.go
package strand

func ToRNA(dna string) string {
	var result string
	for _, dnaNuc := range dna {
		switch dnaNuc {
		case 'G':
			result = result + "C"
		case 'C':
			result = result + "G"
		case 'T':
			result = result + "A"
		case 'A':
			result = result + "U"
		}
	}
	return result
}

File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

to_rna(_Strand) ->
    lists:map(fun from_dna/1, _Strand).

from_dna($G) -> $C;
from_dna($C) -> $G;
from_dna($T) -> $A;
from_dna($A) -> $U.
